### PR TITLE
orig src: cleanups from initial impl

### PR DESCRIPTION
**Pull request by howardjohn**
* Switch to a library impl of Forwarded. The one we wrote does not handle edge cases. Added tests to cover these.
* Unconditionally append Forwarded (still useful, even if its not used by the ztunnel on the other end)
* Unconditionally spoof the outbound IP
* Only use `Forwarded` from trusted waypoint
* Allow explicitly opt in or opt out. By default, auto detect support

This aligns with https://docs.google.com/document/d/17fDTMLg-22UPfh62d5wumAOIpbeoOyWix2WTE54hDNc/edit#

I also wrote e2e tests in istio/istio; they pass with and without this PR

### CODE REVIEW

File: src/proxy.rs [hunk 1]
```diff
@@ -244,35 +243,50 @@ impl TryFrom<&str> for TraceParent {
...
-        first
-            .parse::<IpAddr>()
-            .map_or_else(|_| first.split(':').next()?.parse::<IpAddr>().ok(), Some)
-    }
+    req.headers()
+        .get(header::FORWARDED)
+        .and_then(|rh| rh.to_str().ok())
+        .and_then(|rh| http_types::proxies::Forwarded::parse(rh).ok())
+        .and_then(|ph| {
+            println!("got {:?}", ph.forwarded_for().last());
```
Comment (stevenctl) [inline src/proxy.rs:285]: remove before merging

**Review by stevenctl — APPROVED** [review]
